1. Generate inspector files using genrule_cmd.
2. Set v8_enable_handle_zapping=False to regain performance.
3. Replace @zlib with //external:zlib and fix the include path.
4. Fix build on Windows.

--- BUILD.bazel
+++ BUILD.bazel
@@ -17,6 +17,7 @@ load(
     "v8_torque",
 )
 load(":bazel/v8-non-pointer-compression.bzl", "v8_binary_non_pointer_compression")
+load(":genrule_cmd.bzl", "genrule_cmd")
 
 # =================================================
 # Flags
@@ -104,7 +105,7 @@ v8_flag(name = "v8_enable_disassembler")
 
 v8_flag(
     name = "v8_enable_handle_zapping",
-    default = True,
+    default = False,
 )
 
 v8_flag(name = "v8_enable_hugepage")
@@ -3092,8 +3093,7 @@ genrule(
         "src/inspector/protocol/Schema.cpp",
         "src/inspector/protocol/Schema.h",
     ],
-    cmd = "bazel/generate-inspector-files.sh $(@D)",
-    cmd_bat = "bazel\\generate-inspector-files.cmd $(@D)",
+    cmd = genrule_cmd("@envoy//bazel/external:wee8.genrule_cmd"),
     local = 1,
     message = "Generating inspector files",
 )
@@ -3241,7 +3241,7 @@ v8_library(
     ],
     deps = [
         ":v8_libbase",
-        "@zlib",
+        "//external:zlib",
     ],
 )
 
--- bazel/defs.bzl
+++ bazel/defs.bzl
@@ -367,6 +367,7 @@ def _v8_target_cpu_transition_impl(settings, attr):
         "x86_64": "x64",
         "darwin": "x64",
         "darwin_x86_64": "x64",
+        "x64_windows": "x64",
         "x86": "ia32",
         "ppc": "ppc64",
         "arm64-v8a": "arm64",
--- src/snapshot/snapshot-utils.cc
+++ src/snapshot/snapshot-utils.cc
@@ -5,7 +5,7 @@
 #include "src/snapshot/snapshot-utils.h"
 
 #include "src/base/sanitizer/msan.h"
-#include "third_party/zlib/zlib.h"
+#include "zlib.h"
 
 namespace v8 {
 namespace internal {
